controllers/clusters/cassandra_controller.go

```diff
@@ -738,40 +752,32 @@ func (r *CassandraReconciler) newWatchStatusJob(c *v1beta1.Cassandra) scheduler.
 			return err
 		}
 
-		iCassandra, err := c.FromInstAPI(iData)
-		if err != nil {
-			l.Error(err, "Cannot convert cluster from the Instaclustr API",
-				"cluster name", c.Spec.Name,
-				"cluster ID", c.Status.ID,
-			)
-			return err
-		}
+		iCassandra := v1beta1.Cassandra{}
+		iCassandra.FromInstAPI(instaModel)
 
-		if !areStatusesEqual(&iCassandra.Status.ClusterStatus, &c.Status.ClusterStatus) {
-			l.Info("Updating cluster status",
-				"status from Instaclustr", iCassandra.Status.ClusterStatus,
-				"status from k8s", c.Status.ClusterStatus)
+		if !c.Status.Equals(&iCassandra.Status) {
+			l.Info("Updating cluster status")
 
-			areDCsEqual := areDataCentresEqual(iCassandra.Status.ClusterStatus.DataCentres, c.Status.ClusterStatus.DataCentres)
+			dcsDiffer := c.Status.DCEquals(iCassandra.Status.DataCentres)
 
 			patch := c.NewPatch()
-			c.Status.ClusterStatus = iCassandra.Status.ClusterStatus
+			c.Status.FromInstAPI(instaModel)
 			err = r.Status().Patch(context.Background(), c, patch)
 			if err != nil {
 				return err
 			}
 
-			if !areDCsEqual {
+			if !dcsDiffer {
```

Comment:
it should be differ, like: `if dcsDiffer` -> then do smth. 
or change var name to `dcsEqual`.

Reply:
Done